synapseclient/models/agent.py

```diff
@@ -0,0 +1,447 @@
+        ).start_async(synapse_client=syn)
+        self.sessions[session.id] = session
+        self.current_session = session.id
+        return session
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Agent_Session: {self.registration_id}"
+    )
+    async def get_session_async(
+        self, *, session_id: str, synapse_client: Optional[Synapse] = None
+    ) -> "AgentSession":
+        syn = Synapse.get_client(synapse_client=synapse_client)
+        session = await AgentSession(id=session_id).get_async(synapse_client=syn)
+        if session.id not in self.sessions:
+            self.sessions[session.id] = session
+        self.current_session = session.id
+        return session
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Prompt_Agent_Session: {self.registration_id}"
+    )
+    async def prompt(
+        self,
+        *,
+        session_id: Optional[str] = None,
+        prompt: str,
+        enable_trace: bool = False,
+        newer_than: Optional[int] = None,
+        print_response: bool = False,
+        synapse_client: Optional[Synapse] = None,
+    ) -> None:
+        """Sends a prompt to the agent for the current session.
+            If no session is currently active, a new session will be started.
+
+        Arguments:
+            session_id: The ID of the session to send the prompt to. If None, the current session will be used.
+            prompt: The prompt to send to the agent.
+            enable_trace: Whether to enable trace for the prompt.
+            newer_than: The timestamp to get trace results newer than. Defaults to None (all results).
+            print_response: Whether to print the response to the console.
+            synapse_client: The Synapse client to use for the request. If None, the default client will be used.
+        """
+        syn = Synapse.get_client(synapse_client=synapse_client)
+
+        # TODO: Iron this out. It's a little confusing.
+        if session_id:
+            if session_id not in self.sessions:
+                await self.get_session_async(session_id=session_id, synapse_client=syn)
+            else:
+                self.current_session = session_id
+        else:
+            if not self.current_session:
+                await self.start_session_async(synapse_client=syn)
+
+        await self.sessions[self.current_session].prompt_async(
+            prompt=prompt,
+            enable_trace=enable_trace,
+            newer_than=newer_than,
+            print_response=print_response,
+            synapse_client=syn,
+        )
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Agent_Session_Chat_History: {self.registration_id}"
+    )
+    def get_chat_history(self) -> List[AgentPrompt]:
+        """Gets the chat history for the current session."""
+        # TODO: Is this the best way to do this?
+        return self.sessions[self.current_session].chat_history
```

Comment:
With the suggestion above this could swap to

`self.current_session.chat_history if self.current_session else None`

Reply:
https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1152/files#r1907958583